data-prepper-api/src/test/java/org/opensearch/dataprepper/logging/SensitiveArgumentMaskingConverterTest.java

```diff
@@ -0,0 +1,116 @@
+    @Mock
+    private LogEvent logEvent;
+    @Mock
+    private Message message;
+    @Mock
+    private org.apache.logging.log4j.Marker log4jMarker;
+
+    @Test
+    void testNewInstance() {
+        assertThat(SensitiveArgumentMaskingConverter.newInstance().getName(), equalTo("sensitiveArgument"));
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideSensitiveMarker")
+    void testFormatAppliedToLogEventWithSensitiveMarkers(final Marker marker) {
+        final SensitiveArgumentMaskingConverter objectUnderTest = SensitiveArgumentMaskingConverter.newInstance();
+        when(log4jMarker.getName()).thenReturn(marker.getName());
+        when(logEvent.getMarker()).thenReturn(log4jMarker);
+        when(logEvent.getMessage()).thenReturn(message);
+        when(message.getFormat()).thenReturn("Test log message with anchor: {}");
+        when(message.getParameters()).thenReturn(new Object[] {"value"});
+        final StringBuilder stringBuilder = new StringBuilder();
+        objectUnderTest.format(logEvent, stringBuilder);
+        assertThat(stringBuilder.toString(), equalTo(
+                String.format("Test log message with anchor: %s", MASK_PATTERN)));
+    }
+
+    @Test
+    void testFormatAppliedToLogEventWithOtherMarker() {
+        final SensitiveArgumentMaskingConverter objectUnderTest = SensitiveArgumentMaskingConverter.newInstance();
+        final String testMarkerName = "unknown-" + RandomStringUtils.randomAlphabetic(5);
+        when(log4jMarker.getName()).thenReturn(testMarkerName);
+        when(logEvent.getMarker()).thenReturn(log4jMarker);
+        when(logEvent.getMessage()).thenReturn(message);
+        final String testFormattedMessage = "Test log message";
+        when(message.getFormattedMessage()).thenReturn(testFormattedMessage);
+        final StringBuilder stringBuilder = new StringBuilder();
+        objectUnderTest.format(logEvent, stringBuilder);
+        assertThat(stringBuilder.toString(), equalTo(testFormattedMessage));
+        verify(message, times(0)).getFormat();
+        verify(message, times(0)).getParameters();
+    }
+
+    @Test
+    void testFormatAppliedToLogEventWithoutMarker() {
+        final SensitiveArgumentMaskingConverter objectUnderTest = SensitiveArgumentMaskingConverter.newInstance();
+        when(logEvent.getMarker()).thenReturn(null);
+        when(logEvent.getMessage()).thenReturn(message);
+        final String testFormattedMessage = "Test log message";
+        when(message.getFormattedMessage()).thenReturn(testFormattedMessage);
+        final StringBuilder stringBuilder = new StringBuilder();
+        objectUnderTest.format(logEvent, stringBuilder);
+        assertThat(stringBuilder.toString(), equalTo(testFormattedMessage));
+        verify(message, times(0)).getFormat();
+        verify(message, times(0)).getParameters();
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideSensitiveLogMessageBuildingBlocksAndResult")
```

Comment:
It might be good to include a unit test which also provides an exception as a parameter to the `LogFormat`.